tests/platform_tests/test_kdump.py

```diff
@@ -0,0 +1,117 @@
+import logging
+import time
+import pytest
+
+from tests.common.helpers.assertions import pytest_assert
+from tests.common.platform.processes_utils import wait_critical_processes
+from tests.common.reboot import SONIC_SSH_PORT, SONIC_SSH_REGEX, wait_for_startup, REBOOT_TYPE_COLD
+from tests.platform_tests.test_reboot import check_interfaces_and_services
+
+pytestmark = [
+    pytest.mark.disable_loganalyzer,
+    pytest.mark.topology('any')
+]
+
+SSH_SHUTDOWN_TIMEOUT = 360
+SSH_STARTUP_TIMEOUT = 420
+
+SSH_STATE_ABSENT = "absent"
+SSH_STATE_STARTED = "started"
+
+
+class TestKernelPanic:
+    """
+    This test case is used to verify that DUT will load kdump crashkernel on kernel panic.
+    """
+    def wait_lc_healthy_if_sup(self, duthost, duthosts, localhost, conn_graph_facts, xcvr_skip_list):
+        # For sup, we also need to ensure linecards are back and healthy for following tests
+        is_sup = duthost.get_facts().get("modular_chassis") and duthost.is_supervisor_node()
+        if is_sup:
+            for lc in duthosts.frontend_nodes:
+                wait_for_startup(lc, localhost, delay=10, timeout=300)
+                wait_critical_processes(lc)
+                check_interfaces_and_services(lc, conn_graph_facts["device_conn"][lc.hostname],
+                                              xcvr_skip_list, reboot_type=REBOOT_TYPE_COLD)
+
+    @pytest.fixture(autouse=True)
+    def tearDown(self, duthosts, enum_rand_one_per_hwsku_hostname,
+                 localhost, pdu_controller):
+        yield
+        # If the SSH connection is not established, or any critical process is exited,
+        # try to recover the DUT by PDU reboot.
+        duthost = duthosts[enum_rand_one_per_hwsku_hostname]
+        dut_ip = duthost.mgmt_ip
+        hostname = duthost.hostname
+        if not self.check_ssh_state(localhost, dut_ip, SSH_STATE_STARTED):
+            if pdu_controller is None:
+                logging.error("No PDU controller for {}, failed to recover DUT!".format(hostname))
+                return
+            self.pdu_reboot(pdu_controller)
+            # Waiting for SSH connection startup
+            pytest_assert(self.check_ssh_state(localhost, dut_ip, SSH_STATE_STARTED, SSH_STARTUP_TIMEOUT),
+                          'Recover {} by PDU reboot failed'.format(hostname))
+            # Wait until all critical processes are healthy.
+            wait_critical_processes(duthost)
+            self.wait_lc_healthy_if_sup(duthost, duthosts, localhost)
+
+    def test_kernel_panic(self, duthosts, enum_rand_one_per_hwsku_hostname, localhost,
+                          conn_graph_facts, xcvr_skip_list):
+        duthost = duthosts[enum_rand_one_per_hwsku_hostname]
+        dut_ip = duthost.mgmt_ip
+        hostname = duthost.hostname
+        dut_datetime = duthost.get_now_time()
+
+        out = duthost.command('show kdump config')
+        if "Enabled" not in out["stdout"]:
```

Comment:
do we support the entire behavior in this document - https://github.com/sonic-net/SONiC/blob/master/doc/kdump/SONiC-kdump.md . If so, then why skip the test if disabled, cant we enable kdump with CLI and then proceed with the test -
root@croc-xx36-dut:/home/cisco# config kdump ?
Usage: config kdump [OPTIONS] COMMAND [ARGS]...

Configure the KDUMP mechanism

Options:
-h, -?, --help Show this message and exit.

Commands:
disable Disable the KDUMP mechanism
enable Enable the KDUMP mechanism
memory Configure the memory for KDUMP mechanism
num_dumps Configure the maximum dump files of KDUMP mechanism

Reply:
This test will run only if kdump is already enabled, as we are not sure if other platforms support it.